src/reducers/ProgressTrackerReducer.jsx

```diff
@@ -0,0 +1,67 @@
+const initialState = {
+  genreqs: 0,
+  majorreqs: 0,
+};
+
+function sum(list) {
+  let total = 0;
+  for (let i = 0, len = list.length; i < len; i += 1) {
+    total += list[i].credits;
+  }
+  return total;
+}
+
+const ProgressTrackerReducer = (state = initialState, action) => {
+  const mathCourses = [];
+  const engrCourses = [];
+  const sciCourses = [];
+  const ahseCourses = [];
+
+  switch (action.type) {
+    case 'GET_COMPLETED_COURSES': {
+      const data = action.data;
+      let genreqs = data.map(a => a.generalRequirements[0]);
+      genreqs = genreqs.filter(n => n !== undefined);
+      let majorreqs = data.map(a => a.otherRequirements[0]);
+      majorreqs = majorreqs.filter(n => n !== undefined);
+
+      for (let i = 0; i < data.length; i += 1) {
+        if (data[i].registrarId.substring(0, 3) === 'MTH') {
```

Comment:
Personally I think the syntax of a switch statement looks cleaner.  The problem here is that the number of characters in each substring is not consistent (3 and 4).  So you'd probably have to just substring the first 3 characters and check for `'ENG'` and `'AHS'` instead of `'ENGR'` and `'AHSE'`

In the long term, it makes a lot more sense to have a field for the department in the database, but it might not be worth the time to modify the database at this point.  Up to you if you think that's worth it!